File: solutions/beecrowd/1262/1262.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    std::string line;
    std::int16_t proc, read, clocks;

    while (getline(std::cin, line)) {
        clocks = 0;
        read = 0;
        std::cin >> proc;

        for (size_t i = 0; i < line.length(); ++i) {
            if (line.at(i) == 'W') {
                clocks++;
            } else {
                read++;
            }

            if (line.at(i) == 'W' || i == line.length() - 1) {
                if (read) {
                    clocks += read / proc;
                    if (read % proc != 0) {
                        clocks++;
                    }
                    read = 0;
                }
            }
        }

        std::cout << clocks << std::endl;

        getline(std::cin, line);  // it's a hack
    }
    return 0;
}

File: solutions/beecrowd/1262/1262.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    clocks = 0
    read = 0
    proc = int(input().strip())

    for index, character in enumerate(line):
        if character == 'W':
            clocks += 1
        else:
            read += 1

        if character == 'W' or index == len(line) - 1:
            if read:
                clocks += read // proc
                if not read % proc == 0:
                    clocks += 1
                read = 0

    output(f'{clocks}\n')
